internal/attestation/vsa.go

```diff
@@ -0,0 +1,113 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package attestation
+
+import (
+	"strings"
+
+	"github.com/in-toto/in-toto-golang/in_toto"
+
+	"github.com/enterprise-contract/ec-cli/internal/evaluator"
+	"github.com/enterprise-contract/ec-cli/internal/policy"
+	"github.com/enterprise-contract/ec-cli/internal/policy/source"
+)
+
+const (
+	// Make it visible elsewhere
+	PredicateVSAProvenance = "https://enterprisecontract.dev//verification_summary/v1"
```

Comment:
I don't think we should deviate from the SLSA VSA spec, we'll end up being locked in our own ecosystem

Reply:
My thought is we can produce the SLSA spec and our own, since the results we produce don't conform exactly to what the slsa spec calls for. For instance, we allow for an array of policies where as the slsa spec only allows for one.

Reply:
Let's try to influence the spec first before inventing our own. What we might consider instead is to use the value passed in via `--policy` argument, which is a single string and it's digest.

Reply:
IMO, the SLSA VSA spec as is today is not sufficient for our needs. We may want to produce one, but that's mostly for compatibility with other tools that do expect a SLSA VSA. My understanding is that this is not the requirement as per EC-189, but I could be wrong. For EC-189, I believe the ask was to produce a very EC specific predicate, _inspired_ by SLSA's VSA.

I don't know if it makes sense to try to influence the SLSA VSA to meet our needs. EC handles policies, to meet business requirements, beyond what SLSA provides, e.g. asserting SLSA Level 3 is not sufficient for the use cases where EC is currently used. But it's worth having that conversation with the SLSA community. I just wouldn't block any efforts on that. It's perfectly fine to introduce an experimental predicate type.

A minor commentary is where the code is placed. `internal/attestation` is meant for parsing input predicates, and here we are emitting a predicate. This might be better suited as just another `--output` format.

Reply:
Yep, made more sense under `internal/applicationsnapshot` . thanks!